core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -200,7 +210,16 @@ private TableMetadata getTableMetadata(Scan scan) throws ExecutionException {
     return metadata.getTableMetadata();
   }
 
-  public Optional<List<Key>> get(Scan scan) {
+  public boolean contains(Get get) {
+    return getSet.containsKey(get);
+  }
+
+  public Optional<TransactionResult> get(Get get) {
+    assert this.contains(get);
```

Comment:
This `get` will be called only when creating a result in `CrudHandler` with a guarantee of its existence. So, we should not just return empty since it doesn't mean there is no such record on the database.

Reply:
A "guarantee of its existence" meant the entry of `getSet`. `getSet` is a cache for the results of get operations similar to `scanSet`. So, it might be empty if the record actually does not exist.

Reply:
@feeblefakie Revised in 9cffae7. PTAL again when you get a chance!

Note: I changed `contains` to `containsKey` in the assertion expression to clearly show we are checking the key existence. But I kept `public boolean contains()` to make it different from `containsKeyInReadSet()` that actually checks based on `Key`.

Reply:
@jnmt Thank you!

> But I kept public boolean contains() to make it different from containsKeyInReadSet() that actually checks based on Key.
If you want to keep it, how about naming it `containsKeyInGetSet()` to make it consistent with `containsKeyInReadSet()`?
I feel `contains(get)` is confusing because we assume it checks if the getSet contains the given get instead of the key.

Reply:
@feeblefakie Your concern is definitely right. Fixed in d951832. PTAL again. Thank you!